lib/src/presentation/widget/drawer.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+
+import '../../core/util/constants.dart';
+import 'drawer_list_tile.dart';
+
+class MyDrawer extends StatefulWidget {
+  const MyDrawer({
+    super.key,
+  });
+
+  @override
+  State<MyDrawer> createState() => _MyDrawerState();
+}
+
+enum Route {
+  home(
+    Icons.home,
+    'Home',
+    Constants.homeRouteName,
+  ),
+  popular(
+    Icons.stars_rounded,
+    'Popular Movies',
+    Constants.popularMoviesRouteName,
+  ),
+  upcoming(
+    Icons.arrow_forward_outlined,
+    'Upcoming Movies',
+    Constants.upcomingMoviesRouteName,
+  );
+
+  const Route(
+    this.icon,
+    this.routeName,
+    this.route,
+  );
+
+  final IconData icon;
+  final String routeName;
+  final String route;
+}
```

Comment:
Nice implementation, but move enum to a new file

Reply:
solved